scm/driver/stash/webhook.go

```diff
@@ -100,8 +99,24 @@ func (s *webhookService) parsePullRequest(data []byte) (scm.Webhook, error) {
 	switch src.EventKey {
 	case "pr:opened":
 		dst.Action = scm.ActionOpen
+	case "pr:from_ref_updated":
+		// The request includes field "previousFromHash", which could be compared
+		// to the FromRef.Latestcommit to ensure there is actually a change,
+		// but there is unlikely need for that.
+		dst.Action = scm.ActionSync
+	case "pr:modified":
+		// BitBucket Server (Stash) sends "pr:modified" for any edits to the PR,
+		// including edits to the title or description. Thus, return the hook
+		// action only when the target reference has changed (name or hash).
+		if src.PullRequest.ToRef.DisplayID == src.PreviousTarget.DisplayID &&
+		src.PullRequest.ToRef.LatestCommit == src.PreviousTarget.LatestCommit {
+			return nil, nil
```

Comment:
I presume this is checking to differentiate between title / description change vs a code change?  If this is just a title or description change we can set the action to Updated as opposed to returning a nil.

```diff
		if src.PullRequest.ToRef.DisplayID == src.PreviousTarget.DisplayID &&
		src.PullRequest.ToRef.LatestCommit == src.PreviousTarget.LatestCommit {
+			dst.Action = scm.ActionUpdate
		} else {
			dst.Action = scm.ActionSync
		}

```

Reply:
It is. I'll make that change. I wasn't sure what the `ActionUpdate` was used for.